junit-platform-engine/src/main/java/org/junit/platform/engine/TestDescriptor.java

```diff
@@ -182,6 +182,24 @@ default void accept(Visitor visitor) {
 	}
 
 	/**
+	 * Does a logical level display name resolution for legacy reporting formats, instead of a source level name resolution.
+	 *
+	 * @return A LegacyReportingInfo containing the logical method and class display names.
+	 */
+	default LegacyReportingInfo getLegacyReportingInfo() {
+		final String displayName = getDisplayName();
+		final Optional<String> className = getParent().map(TestDescriptor::getDisplayName);
+
+		return new DefaultLegacyReportingInfo(displayName, className.orElse(null));
+	}
+
+	interface LegacyReportingInfo {
```

Comment:
Please convert this interface into a top-level interface, add Javadoc and an `@API` annotation.